### PR TITLE
Leaflet 1.0 for select on map update

### PR DESCRIPTION
**Pull request by ctwoolsey**

### CODE REVIEW

File: app/scripts/controllers/locationPickerMap.ctrl.js [hunk 1]
```diff
@@ -1,57 +1,192 @@
...
+          fillOpacity: 0
+        }
+      },
+      onEachFeature: onEachFeature
+    }).addTo(map);
+
   });
 
+
+  function muniObj(muniId, muniName){
```
Comment (abobwhite) [inline app/scripts/controllers/locationPickerMap.ctrl.js]: If you need to make a new object (muniObj is probably not the name you want), please define it in a new file. Probably in a "models" or "domain" folder. And I would just call it Municipality.

Comment (abobwhite) [inline app/scripts/controllers/locationPickerMap.ctrl.js]: It just doesn't make sense to define in the map ctrl.

File: app/scripts/controllers/home.ctrl.js [hunk 2]
```diff
@@ -125,7 +139,28 @@ angular.module('yourStlCourts').controller('HomeCtrl', function (Citations,toast
     modalInstance.result.then(function (selectedMunicipalities) {
       ctrl.citationCriteria.municipalityNames = selectedMunicipalities;
     });
+
   };
 
+  function modifyMunicipalities(municip){
```
Comment (abobwhite) [inline app/scripts/controllers/home.ctrl.js]: Is this logic going to be used throughout the app? If so, bury it behind the Municipalities service and just always return the list from the promise with this change. Also - it looks like this is duplicated in your MuniObj I commented on below.

File: app/scripts/controllers/home.ctrl.js [hunk 3]
```diff
@@ -84,7 +87,15 @@ angular.module('yourStlCourts').controller('HomeCtrl', function (Citations,toast
     } else if(optionSelectedMap[ctrl.OptionToSelect.LOCATION]) {
       var names = [];
       ctrl.citationCriteria.municipalityNames.forEach(function(municip){
-        names.push(municip.municipality_name);
+        if (municip.municipality_name == "St. Louis County"){
```
Comment (abobwhite) [inline app/scripts/controllers/home.ctrl.js]: None of this new logic has tests. Also - consider putting this logic behind the Municipalities angular service instead

File: app/scripts/controllers/home.ctrl.js [hunk 4]
```diff
@@ -3,13 +3,16 @@
...
+  //because it will also modify the original municipalties object to remove the "Unincorporated" Key Word
+  ctrl.modifiedMunicipalities = modifyMunicipalities(municipalities);
   ctrl.municipalities = municipalities;
   ctrl.selectedMunicipality = null;
   ctrl.citationCriteria = {};
   ctrl.hasEverSelected = false;
 
   ctrl.municipalitySelected  = function(){
-    $state.go('courtSearchInfo',{courtId : ctrl.selectedMunicipality.court_id});
+    $state.go('courtSearchInfo',{'courtId' : ctrl.selectedMunicipality.court_id});
```
Comment (abobwhite) [inline app/scripts/controllers/home.ctrl.js]: This change doesn't do anything. Typically, in JS, keys don't have string tics around them.

File: app/scripts/controllers/locationPickerMap.ctrl.js (hunk 1 above)
Comment (ctwoolsey) [inline app/scripts/controllers/locationPickerMap.ctrl.js]: The muniObj is not a true municipality object.  It is only used by the map ctrl and specific to it.  It has an id and name field plus a function to help map it to the true municipality object.  The id field that it has has no relation to the database.  It is the id used in the map geoJSON data.

**Pull request merged**